src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -271,8 +271,27 @@ private static function normalizeBody($body)
             return http_build_query($body, '', '&', PHP_QUERY_RFC1738);
         }
 
+        $generatorToCallable = static function (\Generator $body) {
```

Comment:
should we wrapped inside
`if (!\is_string($body)) {`

Reply:
moved `generatorToCallable` Closure to a class static function.